data/src/test/java/org/oppia/android/data/backends/gae/api/ClassroomServiceTest.kt

```diff
@@ -47,6 +49,18 @@ class ClassroomServiceTest {
     assertThat(classroomResponse.body()!!.topicSummaryDicts?.get(0)?.name).isEqualTo("Math")
   }
 
+  @Test
+  fun testMockitoUnitTest() {
+    val mockLicenseFetcher = Mockito.mock(LicenseFetcher::class.java)
+    `when`(mockLicenseFetcher.fetchText("https//www.google.com")).thenReturn("abhay")
+
+    assertThat(mockLicenseFetcher.fetchText("https//www.google.com")).contains("abhay")
+  }
+
+  interface LicenseFetcher {
+    fun fetchText(link: String): String
+  }
+
```

Comment:
(This is only a sample test setup in the data module so that I could explore different options while writing the test case)
I tried mocking the network part with the help of Mockito, and this test case is passing, but according to the documentation, there should be `eq` before the argument of the method while defining the mockito behavior, so the correct implementation should be - 

```kotlin
@Test
  fun testMockitoUnitTest() {
    val mockLicenseFetcher = Mockito.mock(LicenseFetcher::class.java)
    `when`(mockLicenseFetcher.fetchText(eq("https//www.google.com"))).thenReturn("abhay")

    assertThat(mockLicenseFetcher.fetchText("https//www.google.com")).contains("abhay")
  }
```

But this test case fails and the error that I get is  - 

```
java.lang.NullPointerException: eq("https//www.google.com") must not be null

	at org.oppia.android.data.backends.gae.api.ClassroomServiceTest.testMockito(ClassroomServiceTest.kt:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:575)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:263)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
![Screenshot from 2021-07-13 22-17-23](https://user-images.githubusercontent.com/54636525/125501247-e3d8051f-db46-46ad-8f23-832c875ab694.png)

Reply:
You'll probably need to add a new method here to properly handle ``eq``: https://github.com/oppia/oppia-android/blob/develop/testing/src/main/java/org/oppia/android/testing/mockito/MockitoKotlinHelper.kt#L17.

Reply:
Now using mockito-kotlin and the error got fixed.